src/EditorFeatures/CSharpTest/CodeActions/InvertIf/InvertIfTests.Elseless.cs

```diff
@@ -0,0 +1,602 @@
+        {
+            case 1:
+                [||]if (c)
+                {
+                    if (c)
+                    {
+                        return 1;
+                    }
+                }
+
+                return 2;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (!c)
+                {
+                    return 2;
+                }
+
+                if (c)
+                {
+                    return 1;
+                }
+
+                return 2;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless10()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                if (c)
+                {
+                    [||]if (c)
+                    {
+                        return 1;
+                    }
+                }
+                return 2;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                if (c)
+                {
+                    [||]if (!c)
+                    {
+                    }
+                    else
+                    {
+                        return 1;
+                    }
+                }
+                return 2;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless11()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (c)
+                {
+                    f();
+                }
+                g();
+                g();
+                break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (o)
+        {
+            case 1:
+                [||]if (!c)
+                {
+                }
+                else
+                {
+                    f();
+                }
+                g();
+                g();
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElseless12()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        while (c)
+        {
+            if (c)
+            {
+                [||]if (c)
+                {
+                    continue;
+                }
+                if (c())
+                    return;
+            }
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        while (c)
+        {
+            if (c)
+            {
+                [||]if (!c)
+                {
+                    if (c())
+                        return;
+                }
```

Comment:
I have mixed feeligns as well.  Partly this is amazing because it so deeply understands the code and what is/isn't necessary.  i.e. it can tell that moving the 'continue' outside of the first 'if' is unnecessary.  But, in a sense, it feels lossy, and not like i 'inverted' things (where i much more closely intuit that i would want to see code *swapped* around).

Important question, are all these supported cases one that can be run multiple times getting you back (or close) to where you started?  If so, great.  If not... i think my intuition is that we should not offer if we cannot then just let you invokve again to get back (or close to back).

Reply:
In case we introduce `else`, reinvoke will keep it (invertes as a normal swap). again, you can just undo.

Reply:
(bydesign)